libbeat/publisher/processing/default.go

```diff
@@ -20,71 +20,75 @@ package processing
 import (
 	"fmt"
 
+	"github.com/elastic/ecs/code/go/ecs"
+
 	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
 	"github.com/elastic/beats/libbeat/processors"
 	"github.com/elastic/beats/libbeat/processors/actions"
 )
 
-type pipelineProcessors struct {
+// builder is used to create the event processing pipeline in Beats.  The
+// builder orders and merges global and local (per client) event annotation
+// settings, with the configured event processors into one common event
+// processor for use with the publisher pipeline.
+// Also See: (*builder).Create
+type builder struct {
 	info beat.Info
 	log  *logp.Logger
 
 	skipNormalize bool
 
-	// The pipeline its processor settings for
-	// constructing the clients complete processor
-	// pipeline on connect.
+	// global pipeline fields and tags configurations
 	modifiers   []modifier
 	builtinMeta common.MapStr
 	fields      common.MapStr
 	tags        []string
 
-	processors beat.Processor
+	// global pipeline processors
+	processors *group
 
 	drop       bool // disabled is set if outputs have been disabled via CLI
 	alwaysCopy bool
 }
 
-const ecsVersion = "1.0.0-beta2"
-
 type modifier interface {
 	// BuiltinFields defines global fields to be added to every event.
 	BuiltinFields(beat.Info) common.MapStr
 
 	// ClientFields defines connection local fields to be added to each event
-	// of a pipieline client.
+	// of a pipeline client.
 	ClientFields(beat.Info, beat.ProcessingConfig) common.MapStr
 }
 
 type builtinModifier func(beat.Info) common.MapStr
 
-// NewBeatSupport creates a new SupporterFactory based on NewDefaultSupport.
-// NewBeatSupport automatically adds the `ecs.version`, `host.name` and `agent.X` fields
+// MakeDefaultBeatSupport creates a new SupporterFactory based on NewDefaultSupport.
```

Comment:
nit: comment not updated to `SupportFactory`

Reply:
ah, gorename fun :)